In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [20]:
import pandas as pd

# CSV 파일 읽기
expression_df = pd.read_csv("/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/expression_df.csv", index_col=0)
platform_df = pd.read_csv("/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/platform_df.csv", index_col=0)
metadata_df = pd.read_csv("/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/metadata_df.csv", index_col=0)

index_col=0: CSV나 Excel 파일을 pandas로 불러올 때, 첫 번째 열(0번째 열)을 DataFrame의 인덱스로 사용하라는 의미

In [21]:
print(metadata_df.dtypes)

gsm_id         object
subject_id      int64
tissue         object
treatment      object
time_point     object
sex            object
age_yrs         int64
height_cm     float64
weight_kg     float64
bmi_kg/m2     float64
body_fat_%     object
dtype: object


In [22]:
# 수치형으로 변환 (문자열인 경우 처리)
metadata_df["weight_kg"] = pd.to_numeric(metadata_df["weight_kg"], errors="coerce")
metadata_df["body_fat_%"] = pd.to_numeric(metadata_df["body_fat_%"], errors="coerce")

# 체지방량 (kg) 열 추가
metadata_df["pure_fat_kg"] = metadata_df["weight_kg"] * (metadata_df["body_fat_%"] / 100)

In [23]:
metadata_df

gsm_id  subject_id                                tissue  \
0    GSM2062466           1  abdominal subcutaneous white adipose   
1    GSM2062467           1  abdominal subcutaneous white adipose   
2    GSM2062468           1  abdominal subcutaneous white adipose   
3    GSM2062469           2  abdominal subcutaneous white adipose   
4    GSM2062470           2  abdominal subcutaneous white adipose   
..          ...         ...                                   ...   
147  GSM2062613          56  abdominal subcutaneous white adipose   
148  GSM2062614          59  abdominal subcutaneous white adipose   
149  GSM2062615          59  abdominal subcutaneous white adipose   
150  GSM2062616          61  abdominal subcutaneous white adipose   
151  GSM2062617          61  abdominal subcutaneous white adipose   

                 treatment                  time_point     sex  age_yrs  \
0    very-low-calorie diet              at study start  female       46   
1    very-low-calorie diet    after weight loss period  female       46   
2    very-low-calorie diet  after weight stable period  female       46   
3    very-low-calorie diet              at study start  female       45   
4    very-low-calorie diet    after weight loss period  female       45   
..                     ...                         ...     ...      ...   
147       low-calorie diet  after weight stable period    male       56   
148       low-calorie diet              at study start    male       67   
149       low-calorie diet    after weight loss period    male       67   
150  very-low-calorie diet              at study start    male       52   
151  very-low-calorie diet  after weight stable period    male       52   

     height_cm  weight_kg  bmi_kg/m2  body_fat_%  pure_fat_kg  
0        167.0      83.38      29.90        39.9     33.26862  
1        167.0      75.16      26.95        35.8     26.90728  
2        167.0      73.79      26.46        32.4     23.90796  
3        164.0      80.94      30.09        46.7     37.79898  
4        164.0      71.92      26.74        43.0     30.92560  
..         ...        ...        ...         ...          ...  
147      171.0      89.98      30.77        36.5     32.84270  
148      168.5      82.42      29.03        28.2     23.24244  
149      168.5      75.01      26.42        21.2     15.90212  
150      182.0      94.26      28.46        31.1     29.31486  
151      182.0      88.50      26.72        26.6     23.54100  

[152 rows x 12 columns]

### expression_long_df 데이터프레임

In [24]:
expression_df

ID_REF  GSM2062466  GSM2062467  GSM2062468  GSM2062469  GSM2062470  \
0      7892501    4.428059    4.175376    4.025431    4.058749    5.077221   
1      7892502    3.015303    3.382805    4.266405    3.859811    3.524342   
2      7892503    2.469030    1.683800    1.353440    1.833336    1.773266   
3      7892504    7.469548    7.575583    8.319473    7.877318    7.463798   
4      7892505    2.655132    1.349682    1.610853    1.264390    2.165488   
...        ...         ...         ...         ...         ...         ...   
33292  8180414    3.771268    3.827958    4.060053    4.299973    4.432689   
33293  8180415    7.106688    6.772125    6.657147    7.350512    7.153225   
33294  8180416    3.159829    3.384084    3.988990    3.447247    3.426313   
33295  8180417    7.572312    7.260562    7.102751    7.543385    7.276615   
33296  8180418    8.086955    7.506301    7.384701    7.924522    7.381025   

       GSM2062471  GSM2062472  GSM2062473  GSM2062474  ...  GSM2062608  \
0        4.167342    5.217643    5.940471    4.454951  ...    4.535857   
1        3.983906    3.177080    3.634795    2.864439  ...    4.340592   
2        2.073515    2.035071    1.663281    1.647068  ...    2.233719   
3        7.777693    8.129641    7.505768    8.378240  ...    8.569401   
4        1.744837    2.018746    1.284328    1.381185  ...    1.300862   
...           ...         ...         ...         ...  ...         ...   
33292    4.450297    4.603566    4.273023    4.684982  ...    4.710916   
33293    7.043244    7.231078    6.428579    6.299577  ...    6.784532   
33294    3.424887    3.423193    3.636731    3.205910  ...    3.607290   
33295    7.051876    7.267355    6.877710    6.694160  ...    7.217937   
33296    7.339969    7.359909    7.207605    6.963106  ...    7.129784   

       GSM2062609  GSM2062610  GSM2062611  GSM2062612  GSM2062613  GSM2062614  \
0        4.908356    5.564917    2.662255    5.434389    4.318741    5.142372   
1        3.783674    4.117283    3.957558    3.836091    3.138714    4.347246   
2        2.339964    2.191594    2.925588    1.825886    2.154871    2.842330   
3        8.639623    8.087239    8.292656    7.683577    7.317489    7.908107   
4        1.628132    1.477364    1.657482    1.028516    1.138573    1.237365   
...           ...         ...         ...         ...         ...         ...   
33292    4.237877    4.360936    4.887058    4.566632    4.723105    4.759401   
33293    6.962716    6.634560    6.862306    6.961862    6.817866    6.760013   
33294    3.704066    3.750592    3.502732    3.150830    3.621640    3.182532   
33295    6.992108    7.343185    6.941972    7.337702    7.330790    7.255967   
33296    6.373951    6.973372    6.545439    7.355312    6.848314    7.308610   

       GSM2062615  GSM2062616  GSM2062617  
0        5.739793    5.183112    5.107738  
1        4.117830    4.315688    3.758828  
2        2.123143    2.222103    1.485676  
3        7.973787    7.984416    7.821190  
4        1.726360    1.865871    1.279610  
...           ...         ...         ...  
33292    5.170525    4.519322    3.996389  
33293    6.856684    7.077445    7.054198  
33294    3.449176    3.040495    3.080801  
33295    7.227201    7.187702    7.173843  
33296    6.973893    7.526776    6.731832  

[33297 rows x 153 columns]

In [25]:
expression_long_df = expression_df.melt(
    id_vars="ID_REF",
    var_name="gsm_id",       # GSM ID를 열 이름에서 값으로 녹임
    value_name="expression"  # 발현량
)

In [37]:
expression_long_df.columns

Index(['ID_REF', 'gsm_id', 'expression'], dtype='object')

In [36]:
expression_long_df

ID_REF      gsm_id  expression
0        7892501  GSM2062466    4.428059
1        7892502  GSM2062466    3.015303
2        7892503  GSM2062466    2.469030
3        7892504  GSM2062466    7.469548
4        7892505  GSM2062466    2.655132
...          ...         ...         ...
5061139  8180414  GSM2062617    3.996389
5061140  8180415  GSM2062617    7.054198
5061141  8180416  GSM2062617    3.080801
5061142  8180417  GSM2062617    7.173843
5061143  8180418  GSM2062617    6.731832

[5061144 rows x 3 columns]

expression_df['ID_REF']: 유전자 개수가 33297개

metadata_df['gsm_id']: 샘플 개수가 152개

따라서 expression_long의 행은 33297* 152= 5061144개가 나옴!

### expression_change_df 데이터프레임

In [27]:
# expression_long에 metadata 조인 (time_point, subject_id 추가)
expr_meta = expression_long_df.merge(
    metadata_df[['gsm_id', 'subject_id', 'time_point']],
    on='gsm_id',
    how='left'
)

# 피벗: subject_id, ID_REF별로 time_point를 열로 만들기
expr_pivot = expr_meta.pivot_table(
    index=['subject_id', 'ID_REF'],
    columns='time_point',
    values='expression'
).reset_index()

# 열 이름 정리
expr_pivot.columns.name = None
expr_pivot = expr_pivot.rename(columns={
    'after weight stable period': 'C',
    'after weight loss period': 'B',
    'at study start': 'A'
})

# 3. 시간대 간 expression 변화량 계산
expr_pivot['exp_C_A'] = expr_pivot['C'] - expr_pivot['A']
expr_pivot['exp_C_B'] = expr_pivot['C'] - expr_pivot['B']
expr_pivot['exp_B_A'] = expr_pivot['B'] - expr_pivot['A']

expression_change_df = expr_pivot[['subject_id', 'ID_REF', 'exp_C_A', 'exp_C_B', 'exp_B_A']]

In [28]:
expression_change_df

subject_id   ID_REF   exp_C_A   exp_C_B   exp_B_A
0                 1  7892501 -0.402629 -0.149945 -0.252684
1                 1  7892502  1.251101  0.883600  0.367501
2                 1  7892503 -1.115590 -0.330360 -0.785230
3                 1  7892504  0.849925  0.743890  0.106035
4                 1  7892505 -1.044279  0.261170 -1.305449
...             ...      ...       ...       ...       ...
1764736          61  8180414 -0.522933       NaN       NaN
1764737          61  8180415 -0.023247       NaN       NaN
1764738          61  8180416  0.040307       NaN       NaN
1764739          61  8180417 -0.013860       NaN       NaN
1764740          61  8180418 -0.794944       NaN       NaN

[1764741 rows x 5 columns]

33297개의 유전자* 53명의 피실험자= 1764741행을 가짐!

### weight_bmi_changes_df 데이터프레임

In [29]:
metadata_df.columns

Index(['gsm_id', 'subject_id', 'tissue', 'treatment', 'time_point', 'sex',
       'age_yrs', 'height_cm', 'weight_kg', 'bmi_kg/m2', 'body_fat_%',
       'pure_fat_kg'],
      dtype='object')

In [30]:
metadata_df = metadata_df.rename(columns={
    'bmi_kg/m2': 'bmi_kg_m2',
    'body_fat_%': 'body_fat_pct'
})

# 필요한 열만 선택
df = metadata_df[['subject_id', 'time_point', 'weight_kg', 'bmi_kg_m2', 'body_fat_pct', 'pure_fat_kg']]

# time_point 기준으로 pivot
pivoted = df.pivot(index='subject_id', columns='time_point')

# 열 이름 정리
pivoted.columns = ['_'.join(col).strip().replace(' ', '_') for col in pivoted.columns.values]
pivoted = pivoted.reset_index()

In [31]:
# 변화량 계산
pivoted['weight_kg_C_A'] = pivoted['weight_kg_after_weight_stable_period'] - pivoted['weight_kg_at_study_start']
pivoted['weight_kg_C_B'] = pivoted['weight_kg_after_weight_stable_period'] - pivoted['weight_kg_after_weight_loss_period']
pivoted['weight_kg_B_A'] = pivoted['weight_kg_after_weight_loss_period'] - pivoted['weight_kg_at_study_start']

pivoted['bmi_C_A'] = pivoted['bmi_kg_m2_after_weight_stable_period'] - pivoted['bmi_kg_m2_at_study_start']
pivoted['bmi_C_B'] = pivoted['bmi_kg_m2_after_weight_stable_period'] - pivoted['bmi_kg_m2_after_weight_loss_period']
pivoted['bmi_B_A'] = pivoted['bmi_kg_m2_after_weight_loss_period'] - pivoted['bmi_kg_m2_at_study_start']

pivoted['pure_fat_C_A'] = pivoted['pure_fat_kg_after_weight_stable_period'] - pivoted['pure_fat_kg_at_study_start']
pivoted['pure_fat_C_B'] = pivoted['pure_fat_kg_after_weight_stable_period'] - pivoted['pure_fat_kg_after_weight_loss_period']
pivoted['pure_fat_B_A'] = pivoted['pure_fat_kg_after_weight_loss_period'] - pivoted['pure_fat_kg_at_study_start']

# 수치형으로 변환
cols_to_convert = [
    'body_fat_pct_after_weight_stable_period',
    'body_fat_pct_at_study_start',
    'body_fat_pct_after_weight_loss_period'
]
for col in cols_to_convert:
    pivoted[col] = pd.to_numeric(pivoted[col], errors='coerce')  # 숫자로 변환, 오류는 NaN으로 처리

pivoted['fat_C_A'] = pivoted['body_fat_pct_after_weight_stable_period'] - pivoted['body_fat_pct_at_study_start']
pivoted['fat_C_B'] = pivoted['body_fat_pct_after_weight_stable_period'] - pivoted['body_fat_pct_after_weight_loss_period']
pivoted['fat_B_A'] = pivoted['body_fat_pct_after_weight_loss_period'] - pivoted['body_fat_pct_at_study_start']

weight_bmi_changes_df = pivoted[[
    'subject_id',
    'weight_kg_C_A', 'weight_kg_C_B', 'weight_kg_B_A',
    'bmi_C_A', 'bmi_C_B', 'bmi_B_A',
    'fat_C_A', 'fat_C_B', 'fat_B_A',
    'pure_fat_C_A', 'pure_fat_C_B', 'pure_fat_B_A'
]].copy()

In [33]:
weight_bmi_changes_df

subject_id  weight_kg_C_A  weight_kg_C_B  weight_kg_B_A  bmi_C_A  bmi_C_B  \
0            1          -9.59          -1.37          -8.22    -3.44    -0.49   
1            2          -9.97          -0.95          -9.02    -3.70    -0.35   
2            3          -9.47           2.13         -11.60    -2.97     0.68   
3            4          -4.91           0.11          -5.02    -2.07     0.05   
4            5         -12.65          -0.35         -12.30    -3.95    -0.11   
5            6          -3.94           1.87          -5.81    -1.57     0.75   
6            8          -5.66           0.58          -6.24    -2.08     0.21   
7            9          -9.94          -0.57          -9.37    -3.46    -0.20   
8           10         -10.48          -1.02          -9.46    -3.55    -0.35   
9           11          -9.73          -0.73          -9.00    -3.00    -0.22   
10          12          -5.14           0.97          -6.11    -1.89     0.36   
11          13         -11.14          -0.02         -11.12    -3.91    -0.01   
12          14         -10.08           0.67         -10.75    -2.93     0.20   
13          15            NaN          -0.19            NaN      NaN    -0.07   
14          18         -10.42            NaN            NaN    -3.15      NaN   
15          19          -3.93           0.36          -4.29    -1.63     0.15   
16          20          -9.44          -0.96          -8.48    -3.29    -0.34   
17          21          -4.67           0.24          -4.91    -1.86     0.09   
18          22          -5.65           4.06          -9.71    -1.63     1.17   
19          23         -11.85          -0.13         -11.72    -3.74    -0.04   
20          24         -10.14          -0.63          -9.51    -3.57    -0.22   
21          25          -8.41          -0.53          -7.88    -2.73    -0.17   
22          26         -12.16          -1.30         -10.86    -3.42    -0.37   
23          27          -7.62          -0.55          -7.07    -2.65    -0.19   
24          28          -8.92          -2.13          -6.79    -3.34    -0.80   
25          29         -11.34          -2.12          -9.22    -3.28    -0.61   
26          30          -9.31           0.40          -9.71    -2.68     0.11   
27          31          -5.82          -0.88          -4.94    -2.16    -0.33   
28          33          -6.70          -0.46          -6.24    -2.07    -0.14   
29          34          -9.51          -0.80          -8.71    -3.76    -0.32   
30          35          -6.38           0.35          -6.73    -2.22     0.12   
31          36         -10.35          -1.18          -9.17    -3.58    -0.41   
32          37          -8.18          -0.40          -7.78    -2.89    -0.14   
33          38         -11.86          -1.52         -10.34    -4.12    -0.53   
34          39          -3.68           0.40          -4.08    -1.35     0.15   
35          40          -8.43          -1.23          -7.20    -3.38    -0.49   
36          41          -8.67           0.33          -9.00    -3.28     0.13   
37          42         -12.87          -0.18         -12.69    -3.95    -0.05   
38          43         -11.72          -0.16         -11.56    -3.52    -0.05   
39          45          -7.61          -0.35          -7.26    -2.79    -0.13   
40          46         -11.69          -2.27          -9.42    -3.88    -0.75   
41          47          -8.64          -1.46          -7.18    -2.98    -0.50   
42          48          -8.66           0.43          -9.09    -2.64     0.13   
43          49         -10.64          -0.96          -9.68    -3.39    -0.30   
44          50          -4.33           0.06          -4.39    -1.42     0.02   
45          51          -8.82          -0.64          -8.18    -2.85    -0.21   
46          52          -8.13           1.81          -9.94    -2.54     0.57   
47          53            NaN            NaN          -9.28      NaN      NaN   
48          54            NaN      

# SQLite에 데이터 삽입

In [2]:
db_path = "/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/OBESITYGENE_PROJECT.db"

In [39]:
import sqlite3
conn = sqlite3.connect(db_path)

# 테이블 생성
conn.executescript('''
CREATE TABLE expression_change (
    ID_REF TEXT,
    subject_id TEXT,
    exp_C_A REAL,
    exp_C_B REAL,
    exp_B_A REAL,
    PRIMARY KEY (ID_REF, subject_id),
    FOREIGN KEY (ID_REF) REFERENCES platform(ID),
    FOREIGN KEY (subject_id) REFERENCES metadata(subject_id)
);

CREATE TABLE expression_long (
    ID_REF TEXT,
    gsm_id TEXT,
    expression REAL
);

CREATE TABLE platform (
    ID TEXT PRIMARY KEY,
    GB_LIST TEXT,
    SPOT_ID TEXT,
    seqname TEXT,
    RANGE_GB TEXT,
    RANGE_STRAND TEXT,
    RANGE_START INTEGER,
    RANGE_STOP INTEGER,
    total_probes INTEGER,
    gene_assignment TEXT,
    mrna_assignment TEXT,
    category TEXT,
    gene_symbol TEXT
);

CREATE TABLE metadata (
    gsm_id TEXT PRIMARY KEY,
    subject_id TEXT,
    tissue TEXT,
    treatment TEXT,
    time_point TEXT,
    sex TEXT,
    age_yrs REAL,
    height_cm REAL,
    weight_kg REAL,
    bmi_kg_m2 REAL,
    body_fat_pct REAL,
    pure_fat_kg REAL
);

CREATE TABLE weight_bmi_changes (
    subject_id TEXT PRIMARY KEY,

    weight_kg_C_A REAL,
    weight_kg_C_B REAL,
    weight_kg_B_A REAL,

    bmi_C_A REAL,
    bmi_C_B REAL,
    bmi_B_A REAL,

    fat_C_A REAL,
    fat_C_B REAL,
    fat_B_A REAL,

    pure_fat_C_A REAL,
    pure_fat_C_B REAL,
    pure_fat_B_A REAL,

    FOREIGN KEY(subject_id) REFERENCES metadata(subject_id)
);
''')

# 데이터 삽입
platform_df.to_sql("platform", conn, if_exists="append", index=False)
metadata_df.to_sql("metadata", conn, if_exists="append", index=False)
weight_bmi_changes_df.to_sql("weight_bmi_changes", conn, if_exists="append", index=False)
expression_change_df.to_sql("expression_change", conn, if_exists="append", index=False)
expression_long_df.to_sql("expression_long", conn, if_exists="append", index=False)

conn.close()

In [3]:
query1= f"""
-- 열 추가
ALTER TABLE expression_change ADD COLUMN exp_rate_AB REAL;
ALTER TABLE expression_change ADD COLUMN exp_rate_BC REAL;
ALTER TABLE expression_change ADD COLUMN exp_rate_AC REAL;

-- exp_rate_AB = exp_B_A / 기간(B–A)
UPDATE expression_change
SET exp_rate_AB = exp_B_A / (
    SELECT CASE
        WHEN m.treatment = 'LCD' THEN 12
        WHEN m.treatment = 'VLCD' THEN 5
        ELSE NULL
    END
    FROM metadata m
    WHERE m.subject_id = expression_change.subject_id AND m.time_point = 'at study start'
);

-- exp_rate_BC = exp_C_B / 4주 (모두 동일)
UPDATE expression_change
SET exp_rate_BC = exp_C_B / 4.0;

-- exp_rate_AC = exp_C_A / 기간(C–A)
UPDATE expression_change
SET exp_rate_AC = exp_C_A / (
    SELECT CASE
        WHEN m.treatment = 'LCD' THEN 16
        WHEN m.treatment = 'VLCD' THEN 9
        ELSE NULL
    END
    FROM metadata m
    WHERE m.subject_id = expression_change.subject_id AND m.time_point = 'at study start'
);
"""

In [5]:
query2= f"""
ALTER TABLE metadata ADD COLUMN base_weight_kg REAL;
ALTER TABLE metadata ADD COLUMN base_bmi_kg_m2 REAL;
ALTER TABLE metadata ADD COLUMN base_body_fat_pct REAL;
ALTER TABLE metadata ADD COLUMN base_pure_fat_kg REAL;

UPDATE metadata
SET
    base_weight_kg = (
        SELECT m2.weight_kg
        FROM metadata m2
        WHERE m2.subject_id = metadata.subject_id AND m2.time_point = 'at study start'
        LIMIT 1
    ),
    base_bmi_kg_m2 = (
        SELECT m2.bmi_kg_m2
        FROM metadata m2
        WHERE m2.subject_id = metadata.subject_id AND m2.time_point = 'at study start'
        LIMIT 1
    ),
    base_body_fat_pct = (
        SELECT m2.body_fat_pct
        FROM metadata m2
        WHERE m2.subject_id = metadata.subject_id AND m2.time_point = 'at study start'
        LIMIT 1
    ),
    base_pure_fat_kg = (
        SELECT m2.pure_fat_kg
        FROM metadata m2
        WHERE m2.subject_id = metadata.subject_id AND m2.time_point = 'at study start'
        LIMIT 1
    );
    """

In [6]:
query3= f"""
-- ① 열 추가
ALTER TABLE weight_bmi_changes ADD COLUMN weight_kg_rate_AB REAL;
ALTER TABLE weight_bmi_changes ADD COLUMN weight_kg_rate_BC REAL;
ALTER TABLE weight_bmi_changes ADD COLUMN weight_kg_rate_AC REAL;

ALTER TABLE weight_bmi_changes ADD COLUMN bmi_rate_AB REAL;
ALTER TABLE weight_bmi_changes ADD COLUMN bmi_rate_BC REAL;
ALTER TABLE weight_bmi_changes ADD COLUMN bmi_rate_AC REAL;

ALTER TABLE weight_bmi_changes ADD COLUMN fat_rate_AB REAL;
ALTER TABLE weight_bmi_changes ADD COLUMN fat_rate_BC REAL;
ALTER TABLE weight_bmi_changes ADD COLUMN fat_rate_AC REAL;

ALTER TABLE weight_bmi_changes ADD COLUMN pure_fat_rate_AB REAL;
ALTER TABLE weight_bmi_changes ADD COLUMN pure_fat_rate_BC REAL;
ALTER TABLE weight_bmi_changes ADD COLUMN pure_fat_rate_AC REAL;

UPDATE weight_bmi_changes
SET
    weight_kg_rate_AB = weight_kg_B_A / (
        SELECT CASE
            WHEN m.treatment = 'LCD' THEN 12
            WHEN m.treatment = 'VLCD' THEN 5
            ELSE NULL
        END
        FROM metadata m
        WHERE m.subject_id = weight_bmi_changes.subject_id AND m.time_point = 'at study start'
    ),
    weight_kg_rate_BC = weight_kg_C_B / 4.0,
    weight_kg_rate_AC = weight_kg_C_A / (
        SELECT CASE
            WHEN m.treatment = 'LCD' THEN 16
            WHEN m.treatment = 'VLCD' THEN 9
            ELSE NULL
        END
        FROM metadata m
        WHERE m.subject_id = weight_bmi_changes.subject_id AND m.time_point = 'at study start'
    ),

    bmi_rate_AB = bmi_B_A / (
        SELECT CASE
            WHEN m.treatment = 'LCD' THEN 12
            WHEN m.treatment = 'VLCD' THEN 5
            ELSE NULL
        END
        FROM metadata m
        WHERE m.subject_id = weight_bmi_changes.subject_id AND m.time_point = 'at study start'
    ),
    bmi_rate_BC = bmi_C_B / 4.0,
    bmi_rate_AC = bmi_C_A / (
        SELECT CASE
            WHEN m.treatment = 'LCD' THEN 16
            WHEN m.treatment = 'VLCD' THEN 9
            ELSE NULL
        END
        FROM metadata m
        WHERE m.subject_id = weight_bmi_changes.subject_id AND m.time_point = 'at study start'
    ),

    fat_rate_AB = fat_B_A / (
        SELECT CASE
            WHEN m.treatment = 'LCD' THEN 12
            WHEN m.treatment = 'VLCD' THEN 5
            ELSE NULL
        END
        FROM metadata m
        WHERE m.subject_id = weight_bmi_changes.subject_id AND m.time_point = 'at study start'
    ),
    fat_rate_BC = fat_C_B / 4.0,
    fat_rate_AC = fat_C_A / (
        SELECT CASE
            WHEN m.treatment = 'LCD' THEN 16
            WHEN m.treatment = 'VLCD' THEN 9
            ELSE NULL
        END
        FROM metadata m
        WHERE m.subject_id = weight_bmi_changes.subject_id AND m.time_point = 'at study start'
    ),

    pure_fat_rate_AB = pure_fat_B_A / (
        SELECT CASE
            WHEN m.treatment = 'LCD' THEN 12
            WHEN m.treatment = 'VLCD' THEN 5
            ELSE NULL
        END
        FROM metadata m
        WHERE m.subject_id = weight_bmi_changes.subject_id AND m.time_point = 'at study start'
    ),
    pure_fat_rate_BC = pure_fat_C_B / 4.0,
    pure_fat_rate_AC = pure_fat_C_A / (
        SELECT CASE
            WHEN m.treatment = 'LCD' THEN 16
            WHEN m.treatment = 'VLCD' THEN 9
            ELSE NULL
        END
        FROM metadata m
        WHERE m.subject_id = weight_bmi_changes.subject_id AND m.time_point = 'at study start'
    );
    """

In [9]:
import sqlite3

conn = sqlite3.connect(db_path)
cursor = conn.cursor()

cursor.executescript(query1)
cursor.executescript(query2)
cursor.executescript(query3)

conn.commit()
conn.close()

베이스, 변화율 추가 전

In [40]:
conn = sqlite3.connect(db_path)

# 1. 전체 테이블 목록 확인
tables_df = pd.read_sql_query("SELECT name FROM sqlite_master WHERE type='table';", conn)
print("📦 전체 테이블 목록:")
print(tables_df)

# 2. 각 테이블별 컬럼 구조 출력
for table_name in tables_df['name']:
    print(f"\n📄 테이블: {table_name}")
    columns_df = pd.read_sql_query(f"PRAGMA table_info({table_name});", conn)
    print(columns_df)

conn.close()

📦 전체 테이블 목록:
                 name
0   expression_change
1     expression_long
2            platform
3            metadata
4  weight_bmi_changes

📄 테이블: expression_change
   cid        name  type  notnull dflt_value  pk
0    0      ID_REF  TEXT        0       None   1
1    1  subject_id  TEXT        0       None   2
2    2     exp_C_A  REAL        0       None   0
3    3     exp_C_B  REAL        0       None   0
4    4     exp_B_A  REAL        0       None   0

📄 테이블: expression_long
   cid        name  type  notnull dflt_value  pk
0    0      ID_REF  TEXT        0       None   0
1    1      gsm_id  TEXT        0       None   0
2    2  expression  REAL        0       None   0

📄 테이블: platform
    cid             name     type  notnull dflt_value  pk
0     0               ID     TEXT        0       None   1
1     1          GB_LIST     TEXT        0       None   0
2     2          SPOT_ID     TEXT        0       None   0
3     3          seqname     TEXT        0       None   0
4     4 

베이스, 변화율 추가 후

In [11]:
import pandas as pd
conn = sqlite3.connect(db_path)

# 1. 전체 테이블 목록 확인
tables_df = pd.read_sql_query("SELECT name FROM sqlite_master WHERE type='table';", conn)
print("📦 전체 테이블 목록:")
print(tables_df)

# 2. 각 테이블별 컬럼 구조 출력
for table_name in tables_df['name']:
    print(f"\n📄 테이블: {table_name}")
    columns_df = pd.read_sql_query(f"PRAGMA table_info({table_name});", conn)
    print(columns_df)

conn.close()

📦 전체 테이블 목록:
                 name
0   expression_change
1     expression_long
2            platform
3            metadata
4  weight_bmi_changes

📄 테이블: expression_change
   cid         name  type  notnull dflt_value  pk
0    0       ID_REF  TEXT        0       None   1
1    1   subject_id  TEXT        0       None   2
2    2      exp_C_A  REAL        0       None   0
3    3      exp_C_B  REAL        0       None   0
4    4      exp_B_A  REAL        0       None   0
5    5  exp_rate_AB  REAL        0       None   0
6    6  exp_rate_BC  REAL        0       None   0
7    7  exp_rate_AC  REAL        0       None   0

📄 테이블: expression_long
   cid        name  type  notnull dflt_value  pk
0    0      ID_REF  TEXT        0       None   0
1    1      gsm_id  TEXT        0       None   0
2    2  expression  REAL        0       None   0

📄 테이블: platform
    cid             name     type  notnull dflt_value  pk
0     0               ID     TEXT        0       None   1
1     1          GB_LIST  

In [3]:
import sqlite3

# DB 연결
conn = sqlite3.connect(db_path)
cursor = conn.cursor()

# 테이블 이름 가져오기
cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")
tables = cursor.fetchall()

# 각 테이블의 행 수 출력
for table_name in tables:
    table = table_name[0]
    cursor.execute(f"SELECT COUNT(*) FROM {table}")
    row_count = cursor.fetchone()[0]
    print(f"📊 {table}: {row_count} rows")

conn.close()

📊 expression_change: 1764741 rows
📊 expression_long: 5061144 rows
📊 platform: 33297 rows
📊 metadata: 152 rows
📊 weight_bmi_changes: 53 rows
